Source/Particles/Gather/FieldGather.H

```diff
@@ -16,413 +16,305 @@
-                     const amrex::IndexType by_type,
-                     const amrex::IndexType bz_type,
+                     const amrex::ParticleReal zp,
+                     amrex::ParticleReal& Fxp,
+                     amrex::ParticleReal& Fyp,
+                     amrex::ParticleReal& Fzp,
+                     amrex::Array4<amrex::Real const> const& Fx_arr,
+                     amrex::Array4<amrex::Real const> const& Fy_arr,
+                     amrex::Array4<amrex::Real const> const& Fz_arr,
+                     const amrex::IndexType Fx_type,
+                     const amrex::IndexType Fy_type,
+                     const amrex::IndexType Fz_type,
                      const amrex::XDim3 & dinv,
                      const amrex::XDim3 & xyzmin,
                      const amrex::Dim3& lo,
                      [[maybe_unused]] const int n_rz_azimuthal_modes)
 {
     using namespace amrex;
-
     constexpr int zdir = WARPX_ZINDEX;
     constexpr int NODE = amrex::IndexType::NODE;
     constexpr int CELL = amrex::IndexType::CELL;
 
-    // --- Compute shape factors
-
-    Compute_shape_factor< depos_order > const compute_shape_factor;
-    Compute_shape_factor<depos_order - galerkin_interpolation > const compute_shape_factor_galerkin;
-
-#if (AMREX_SPACEDIM >= 2)
-    // x direction
-    // Get particle position
-#ifdef WARPX_DIM_RZ
-    const amrex::Real rp = std::sqrt(xp*xp + yp*yp);
-    const amrex::Real x = (rp - xyzmin.x)*dinv.x;
-#else
-    const amrex::Real x = (xp-xyzmin.x)*dinv.x;
-#endif
-
-    // j_[eb][xyz] leftmost grid point in x that the particle touches for the centering of each current
-    // sx_[eb][xyz] shape factor along x for the centering of each current
-    // There are only two possible centerings, node or cell centered, so at most only two shape factor
-    // arrays will be needed.
-    amrex::Real sx_node[depos_order + 1];
-    amrex::Real sx_cell[depos_order + 1];
-    amrex::Real sx_node_galerkin[depos_order + 1 - galerkin_interpolation] = {0._rt};
-    amrex::Real sx_cell_galerkin[depos_order + 1 - galerkin_interpolation] = {0._rt};
-
-    int j_node = 0;
-    int j_cell = 0;
-    int j_node_v = 0;
-    int j_cell_v = 0;
-    if ((ey_type[0] == NODE) || (ez_type[0] == NODE) || (bx_type[0] == NODE)) {
-        j_node = compute_shape_factor(sx_node, x);
-    }
-    if ((ey_type[0] == CELL) || (ez_type[0] == CELL) || (bx_type[0] == CELL)) {
-        j_cell = compute_shape_factor(sx_cell, x - 0.5_rt);
-    }
-    if ((ex_type[0] == NODE) || (by_type[0] == NODE) || (bz_type[0] == NODE)) {
-        j_node_v = compute_shape_factor_galerkin(sx_node_galerkin, x);
-    }
-    if ((ex_type[0] == CELL) || (by_type[0] == CELL) || (bz_type[0] == CELL)) {
-        j_cell_v = compute_shape_factor_galerkin(sx_cell_galerkin, x - 0.5_rt);
+    // Compute particle position in grid units
+#if defined(WARPX_DIM_RZ)
+    amrex::Real const rp = std::sqrt(xp*xp + yp*yp);
+    amrex::Real costheta, sintheta;
```

Comment:
In other places, this bit of code is written like this, which is cleaner and allows use of `const`.
```
    amrex::Real const costheta = (rp > 0. ? xp/rp : 1.);;
    amrex::Real const sintheta = (rp > 0. ? yp/rp : 0.);;
```